### PR TITLE
convert rails only API app to normal rails app

### LINKED ISSUE #17: Switch the application from API mode to a regular rails app with views
---
name: 🚀 Chore
---

**Context**

We're currently considering switching back to a regular Rails setup as opposed to using API mode + React. I think it would be faster for us to take this approach as we might not have a lot of people working on the project who are familiar with React. This approach could still favor the backend/frontend split if we want to go ahead of this

**Path to Migrate**
Going by this stackoverflow link - https://stackoverflow.com/questions/36669981/how-do-you-convert-a-rails-5-api-app-to-a-rails-app-that-can-act-as-both-api-and

* cd into the app's parent directory
* run `rails new openemrs`
* It'd add files which were not currently in the project and list conflicts if any
* For conflicts:
  * Submit `d` on each one of them to see the differences
  * If the difference is only adding lines, then allow it with `Y`.
  * If the difference is only removing code, then reject it with `n`
  * If the difference is both adding and removing code, write down that file to come back to after this process. Then accept it with `Y`.

* Add a homepage and set it as `root`
* Move controllers out of the `api/v1` subfolders

**Considerations**
* CSS framework to use. Leaning towards Bootstrap since it's more common. Open to other suggestions

### PR DESCRIPTION
**Pull request by abruzy**
#### What does this PR do?

This PR converts the former app which was rails only API to the regular rails app

#### Description of Task to be completed?

- converted the app to the regular rails app

#### How should this be manually tested?

- start the app by running rails s

NB: please replace "your" to the appropriate value

#### Any background context you want to provide?

- Nil

closes #17

### CODE REVIEW

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#   # Specify URI for violation reports
+#   # policy.report_uri "/csp-violation-report-endpoint"
+# end
+
+# If you are using UJS then enable automatic nonce generation
+# Rails.application.config.content_security_policy_nonce_generator = -> request { SecureRandom.base64(16) }
+
+# Set the nonce only to specific directives
+# Rails.application.config.content_security_policy_nonce_directives = %w(script-src)
```
[the remark this answers is not in the document]
Comment (robocopkaka) [reply config/initializers/content_security_policy.rb]: @abruzy Please revert this as we'll need it later

File: config/credentials.yml.enc [hunk 2]
```diff
@@ -1 +1 @@
-X5sExIW75pfjZGQlTsYqUJHADhme9Tr//iSvvlN3qpIiBb98oaBazZR5GZmQJfyBfQSmQSSn4qXT7AKRnLcawQQIgiFXxRorE5P2jNrbV4sMCfpkIZL4db1Xa0tBx4Oga4NPzfJZdgWN+rGkEn9JfVaFPM8MyOzUjezBQkx6AFTdMliEv2+KeaGxqzZgiz7PYwYPQGcVLbMkjeZ69d9iFpN7mMo3DGuuVzFhvjDn5b4INOmtc9gaGSHX6fv2xLY6RT66fK6iWC/SqIDlXlqHiIP3kqHZYTZvFXcLXiS5eTWfct9+vbDApMkj2RbbGabZJpiIQn+W2vxy/fl6r1Ia93hNCI06r2ECO6iCSseRjSxYqvBT8rJV6pjrF4RLcs/eonjUO3pmaT/d1NXG+kLNu/z8PO8DOi9gwql2bfRs/gTxV472DeTZgFJAV6m3ZHhfHJe++eyNouaPXiYLIjY=--55kbFZc4UPP7Eywq--H+M1F+4WCADZzcXkm/GkOw==
\ No newline at end of file
+DB/Aka+J65Dpu9tcX+iWm1wzwzyqwGHg5SXCWEaUMcyaHdphiA1JIKmvAF5I/BVkhlEACs1NFrmAda3XSq/YGYRZG/iT7QuoqASrn+vXUFR70LE9DEKpq+2KM4h+tT57HaG3IytipkYIcWg+ygwW1Yz9kZdMSbrJ7Jjmp+Bn+0bGG7KvFMDOlQ29K9TTrBqPv99jQhAW2IiKXqVAUUdPuvmM0PvRcRQc75VHPh07qmnEByQVixAMVJZm2aartqZpgJ5zzYkVDL8Wk+YjzIPEB8TPhe8rMhnHbZDOM6qBdhbHkgYUO2qr1+oT4Kh+ba32a+0tCILV6sqF3y8A+GfGfmevhgfEs0ZokdVeKXGNVy44TjcTpO57DuQ4qvaBwU8aRywb/sk/3LU/5QLp8fEwKamQMljpOSbAhTc2MqxtAMYXXJ0kG/WWxXYC9e2sUc2Xx81O3Gg0LmzQhxKW+mbXopLqBDcZHw==--8fzTmCf1QygJZHTs--rC5khi8tu5kkTdeyY7KvAA==
```
Comment (robocopkaka) [inline config/credentials.yml.enc:1]: @igbanam This change uses my master key which I should share on Slack or another secure platform

File: config/initializers/cors.rb [hunk 3]
```diff
@@ -13,4 +13,4 @@
 #       headers: :any,
 #       methods: [:get, :post, :put, :patch, :delete, :options, :head]
 #   end
-# end
+# end
```
Comment (robocopkaka) [inline config/initializers/cors.rb:16]: Can you add a new line here

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
-class ApplicationController < ActionController::API
+class ApplicationController < ActionController::Base
+end
+
+class ApiController < ActionController::API
   include V1::Response
   include Error::ErrorHandler
 end
```
Comment (robocopkaka) [inline app/controllers/application_controller.rb:7]: Please move this to a separate file in the `api` subfolder.

File: app/helpers/statics_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module StaticsHelper
```
Comment (robocopkaka) [inline app/helpers/statics_helper.rb:1]: Please rename to `homes`

File: spec/helpers/statics_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe StaticsHelper, type: :helper do
```
Comment (robocopkaka) [inline spec/helpers/statics_helper_spec.rb:3]: Please rename to `homes`. Also the file name

File: spec/requests/statics_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "Statics", type: :request do
```
Comment (robocopkaka) [inline spec/requests/statics_request_spec.rb:3]: Also this

**Review by igbanam — APPROVED** [review]

**Review by robocopkaka — APPROVED** [review]
LGTM

**Pull request merged**